### ISSUE TITLE
[Android] Assets with the same name and different extension don't work properly [7.0]

### ISSUE THREAD

**Issue by MartinZikmund** [unoplatform/uno#9618]
### Current behavior

Suppose the `Assets` folder in Shared project contains the following:

- Image.png
- Image.jpg

Even though both are different actual files, when used in `<Image>` they are retrieved by `DrawableHelper` using their "id" which is using `GetFileNameWithoutExtension`. As a result, only one will display.

### Expected behavior

Should allow displaying multiple images with the same name and different extension.

### Workaround

Use different image name.

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia, Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### Anything else we need to know?

This can be fixed by adjusting the assets retarget task to include the file extension in the file name (e.g. drawables will contain something like `Assets_Image_jpg.jpg` and `Assets_Image_png.png`). Then we can use the filename with extension to get the image

**Comment by MartinZikmund** [comment]
This is still relevant for Skia Android

**Comment by MartinZikmund** [comment]
# Repro generator

This issue was classified as not reproducible with a code sample or test.

**Reason:** Android-specific asset naming bug.

**Comment by MartinZikmund** [comment]
**Audit result: NOT STARTED** in `feature/breakingchanges` (as of `0950717a0d7`).

The Android drawable key-collision bug the issue reports is unfixed on the branch — the lookup and retarget code are byte-identical to `origin/master`, only relocated by the unrelated `Uno.UWP` → `Uno.WinRT` project move. The fix exists as a stalled draft PR (#15891, targets `master`, blocked on a nine-patch regression) and is tracked, unchecked, as `BC74` in the breaking-changes rollup.

### Remaining

- [ ] **Drawable lookup still strips the extension before resolving.** `src/Uno.WinRT/Helpers/DrawableHelper.Android.cs:48` (`FindResourceId`) computes its lookup key via `AndroidResourceNameEncoder.Encode(System.IO.Path.GetFileNameWithoutExtension(imageName))`, and `:87` (`FindResourceIdFromPath`) does the same via `Path.GetFileNameWithoutExtension(AndroidResourceNameEncoder.EncodeDrawablePath(imagePath))` — extension is discarded either way, so `Image.png` and `Image.jpg` still resolve to the same key. `diff <(git show origin/master:src/Uno.UWP/Helpers/DrawableHelper.Android.cs) src/Uno.WinRT/Helpers/DrawableHelper.Android.cs` returns no differences; `git log --oneline --follow` on the current path shows the only branch-side event is `f0f1b2ae2a1 refactor: Move the WinRT project folder to src/Uno` / `c5cf7b5c32d refactor: Move Uno.UWP project folder to Uno.WinRT` — a pure rename, no logic change.
- [ ] **Retarget task still writes the base name without disambiguating by extension.** `src/SourceGenerators/Uno.UI.Tasks/ResourceConverters/AndroidResourceConverter.cs:28` builds the physical file via `AndroidResourceNameEncoder.EncodeDrawablePath(resourceCandidate.LogicalPath)`, and `EncodeDrawablePath` (`src/Uno.WinRT/Helpers/AndroidResourceNameEncoder.cs:66-67`) only encodes the directory/filename and appends the extension verbatim (`encodedFileName + extension`) — it does not fold the extension into the base name the way the issue's own suggested fix requires (e.g. `Assets_Image_jpg.jpg`), so two same-named, differently-extensioned assets in `drawable/` still collide once Android resource compilation (aapt) keys by base name. This file is also unchanged vs. `origin/master:src/Uno.UWP/Helpers/AndroidResourceNameEncoder.cs` (`diff` returns no differences beyond the rename).
- [ ] **Fix PR is a stalled draft, not merged.** `gh pr view 15891` → `{"isDraft":true,"mergeCommit":null,"mergedAt":null,"state":"OPEN"}`, targeting `master` (not `feature/breakingchanges`), body reads "part of #15866, closes #9618". `git merge-base --is-ancestor` is moot since no merge commit exists to check.
- [ ] **Not moot under Skia-only either.** `specs/050-breaking-changes-rollup/spec.md:241-243` lists this as unchecked `BC74` ("Android drawable extension in retarget keys"), explicitly noting: "Deferred out of Phase 5: draft PR #15891 still targets `master` and is blocked on the nine-patch (`.9.png`) regression it exposes. Not moot under Skia-only — Skia-on-Android still resolves drawables through this path." `DrawableHelper` is confirmed live (not dead code): the source generator wires it at `src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/AndroidResourcesGenerator.cs:80` (`global::Uno.Helpers.DrawableHelper.SetDrawableResolver(...)`), and it's also consumed by `src/Uno.WinRT/Storage/StorageFile.Android.cs` and `src/Uno.WinRT/UI/StartScreen/Extensions/JumpListItemExtensions.Android.cs`.
- [ ] **No migration-guide or `PackageDiffIgnore` entry** — this is a bug fix, not an API removal, so neither is expected; a full-text `grep -li` for "drawable" across `build/PackageDiffIgnore.xml` and `doc/articles/migrating-to-uno-7.md` returns nothing, consistent with no behavior change having shipped.

### Caveats (not blocking)

- The issue body itself proposes the exact fix (fold the extension into the encoded key, e.g. `Assets_Image_jpg.jpg`) — this is what #15891 implements, but its own regression (nine-patch `.9.png` handling) is the reason it never merged even to `master`, let alone this branch.
- The issue carries a `no-repro` label from an automated repro-generator pass, and a 2026 comment from Martin Zikmund ("This is still relevant for Skia Android") confirming it is neither stale nor obsolete.

_Audit is code inspection against the branch worktree; no build or runtime validation was performed._
